Cannot compile empty filenames

**V version:** V 0.1.20 e72fe25
**OS:** macOS 10.14

**What did you do?**
```
$ v .v
ld: can't open output file for writing: , errno=2 for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
V error: C error. This should never happen. Please create a GitHub issue: https://github.com/vlang/v/issues/new/choose
```


**What did you expect to see?**

Maybe an `a.out` or `main`? or a more meaningful error message 

**What did you see instead?**

trying to compile some equivalently named ".c" file gives this :
```
$ cat .c
int main() {
	return 0;
}
$ gcc .c
/usr/bin/ld:.c: file format not recognized; treating as linker script
/usr/bin/ld:.c:1: syntax error
collect2: error: ld returned 1 exit status
```
So I would say that it should not be an issue if V also refuses such kind of bogus file name

Works perfectly fine with clang

```
$ cat .c
main() {}
$ clang .c
.c:1:1: warning: type specifier missing, defaults to 'int' [-Wimplicit-int]
main() {}
^
1 warning generated.
$
```

--- a/compiler/main.v
+++ b/compiler/main.v
@@ -899,7 +899,7 @@ fn new_v(args[]string) &V {
 		exit(1)
 	}
 	// No -o provided? foo.v => foo
-	if out_name == 'a.out' && dir.ends_with('.v') {
+	if out_name == 'a.out' && dir.ends_with('.v') && dir != '.v' {
 		out_name = dir.left(dir.len - 2)
 	}
 	// if we are in `/foo` and run `v .`, the executable should be `foo`

